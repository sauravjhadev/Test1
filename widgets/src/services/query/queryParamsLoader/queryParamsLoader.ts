import * as fs from 'fs';

import WidgetFileType, { FileLoaderResponse } from '../../../types';
import { messages } from '../../../messages';

export default function queryParamsLoader(widgetDir: string): Promise<FileLoaderResponse> {
    return new Promise((resolve, reject) => {
        fs.readFile(
            `${widgetDir}/${WidgetFileType.QUERY_PARAMS}`,
            'utf8',
            (error, data: string) => {
                if (error) {
                    if (error.code === 'ENOENT') {
                        resolve({
                            type: WidgetFileType.QUERY_PARAMS,
                            data: '{}',
                        });
                    } else {
                        reject(messages.invalidQueryParams());
                    }
                }

                resolve({
                    type: WidgetFileType.QUERY_PARAMS,
                    data,
                });
            },
        );
    });
}
