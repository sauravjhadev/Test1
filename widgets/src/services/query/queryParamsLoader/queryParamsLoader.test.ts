import queryParamsLoader from './queryParamsLoader';

describe('queryParamsLoader', () => {
    it('correctly loads query params', (done) => {
        queryParamsLoader('src/services/__fixtures__').then((data) => {
            expect(data).toMatchSnapshot();
            done();
        });
    });
});
