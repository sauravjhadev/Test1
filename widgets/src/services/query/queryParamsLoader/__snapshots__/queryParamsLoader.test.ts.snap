// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryParamsLoader correctly loads query params 1`] = `
Object {
  "data": "{
    \\"productIds\\": [1, 2],
    \\"product\\": 1,
    \\"name\\": \\"Fred\\"
}
",
  "type": "queryParams.json",
}
`;
